### PR TITLE
[WIP] Inspect network state before/after docker-py test

### PR DESCRIPTION
**Pull request by arkodg**
relates to https://github.com/moby/moby/issues/39570

### CODE REVIEW

File: hack/make/test-docker-py [hunk 1]
```diff
@@ -45,11 +44,12 @@ source hack/make/.integration-test-helpers
 			exec docker build ${build_opts} -t ${docker_py_image} -f tests/Dockerfile "https://github.com/docker/docker-py.git#${DOCKER_PY_COMMIT}"
```
Comment (thaJeztah) [inline hack/make/test-docker-py]: note that (trying to solve the issue) I configured the build to run with `--network=host`, so the docker build is running in the same networking namespace as the "outer" container.

perhaps we 'll have to check DNS configuration as well?

**Comment by arkodg** [comment]
```
15:55:19 W: Failed to fetch http://deb.debian.org/debian/dists/jessie/InRelease  
15:55:19 
15:55:19 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/InRelease  
15:55:19 
15:55:19 W: Failed to fetch http://security.debian.org/debian-security/dists/jessie/updates/Release.gpg  Temporary failure resolving 'security.debian.org'
15:55:19 
15:55:19 W: Failed to fetch http://deb.debian.org/debian/dists/jessie/Release.gpg  Temporary failure resolving 'deb.debian.org'
15:55:19 
15:55:19 W: Some index files failed to download. They have been ignored, or old ones used instead.
15:55:19 Reading package lists...
15:55:21 Building dependency tree...
15:55:21 Reading state information...
15:55:21 Package gnupg2 is not available, but is referred to by another package.
15:55:21 This may mean that the package is missing, has been obsoleted, or
15:55:21 is only available from another source
15:55:21 
15:55:21 E: Package 'gnupg2' has no installation candidate
15:55:21 E: Unable to locate package pass
```

These errors are seen during `docker build` , has something recently changed in the Dockerfile for docker-py

**Comment by thaJeztah** [comment]
No, that's the problem to look for; those failures show up when running docker-py after the other tests, but doesn't show when running them before

**Comment by arkodg** [comment]
It's probably because .dockerenv gets created during one of the tests and --net=host doesn't get applied causing the build to run in a container with incorrect DNS settings

**Comment by thaJeztah** [comment]
it failed both with `--net=host` and without

**Comment by arkodg** [comment]
since `docker build` uses the daemon's DNS settings iis there a simple way to add `--dns 8.8.8.8` to the daemon settings before this test starts ?

**Comment by arkodg** [comment]
fixed the docker build error but there is more state that has not been cleaned from the previous test
```
docker.errors.APIError: 400 Client Error: Bad Request ("could not choose an IP address to advertise since this system has multiple addresses on different interfaces (172.20.0.1 on br-2d2bd8c8c267 and 172.23.0.1 on br-v5patmwggag6)")
```

**Pull request closed without merging**